src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -249,16 +249,48 @@ async function createEmscriptenModuleInstance(resourceLoader: WebAssemblyResourc
   const existingPostRun = moduleConfig.postRun || [];
   (moduleConfig as any).preloadPlugins = [];
 
+  var totalResources = Object.keys(resources.assembly).length + Object.keys(resources.pdb || {}).length + /* dotnet.wasm */ 1;
+
+  if (resourceLoader.bootConfig.resources.runtime.hasOwnProperty('dotnet.timezones.blat')){
+    totalResources++;
+  }
+
+  if (resourceLoader.bootConfig.icuDataMode !== ICUDataMode.Invariant){
+    totalResources++;
+  }
+
+  const targetElement = document.getElementById('app') as HTMLDivElement;
+  const progressElement = document.getElementById('progress') as unknown as SVGCircleElement;
+  const percentageElement = document.getElementById('percentage') as unknown as SVGTextElement;
```

Comment:
I think we could avoid hardcoding any IDs. In particular, we *can't* hardcode `app`, since there's no requirement for people to load their app into an element with ID `app` (that's just something the default project template happens to do).

I think we'll need to set the CSS variables on `document.documentElement` since we don't know what element the app is going to render things into until the .NET code starts up. As long as people are only starting up one Blazor instance (which is the only supported scenario today), it will be fine for the CSS variables to cascade down from `document.documentElement`.

In the future, if we want to support loading multiple runtime instances and having independent progress reports for each one, we'd have to have some new option on `WebAssemblyStartOptions` to give an element or CSS selector that will contain the "loading" UI. But that's not a scenario today so leave it until it is :)

Reply:
To be fair, if we wanted to, we could also get this values via CSS variables using `
```
document.getElementById(`getComputedStyle(document.documentElement).getPropertyValue('--blazor-loader-target'))
document.getElementById(getComputedStyle(document.documentElement).getPropertyValue('--blazor-loader-progress'))
document.getElementById(getComputedStyle(document.documentElement).getPropertyValue('--blazor-loader-percentage'))
```